sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs

```diff
@@ -42,26 +49,186 @@ pub fn tendermint_state_proof(
     todo!()
 }
 
-pub fn tendermint_init_lc(
+// TODO: remove trace and add document for this funciton.
+#[instrument(level = "trace", skip_all, err)]
+pub fn tendermint_verify_lc(
     context: &mut NativeContext,
     ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
-    todo!()
+    assert!(args.len() == 4);
+    assert!(ty_args.len() == 0);
+
+    let header = pop_arg!(args, Vector).to_vec_u8()?;
+    let commitment_root = pop_arg!(args, Vector).to_vec_u8()?;
+    let next_validators_hash = pop_arg!(args, Vector).to_vec_u8()?;
+    let timestamp = pop_arg!(args, Vector).to_vec_u8()?;
+
+    let Ok(any) = Any::decode(&mut header.as_slice()) else {
+        return Ok(NativeResult::err(context.gas_used(), INVALID_INPUT));
+    };
+
+    let Ok(header) = TmHeader::try_from(any) else {
+        return Ok(NativeResult::err(context.gas_used(), INVALID_INPUT));
```

Comment:
shall we have more specific error codes?
* invalid_protobuf_any
* invalid_tm_header
* invalid_timestamp
* ...

Reply:
Yes, I will make todo and update it in other PR.